### ISSUE TITLE
Unable to use Yourls from subfolder

### ISSUE THREAD

**Issue by ehguille** [YOURLS/YOURLS#3210]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/YOURLS/.github/blob/main/CODE_OF_CONDUCT.md).

### Submission validity

- [X] This is not a personal support request, that should be posted on [discussions community](https://github.com/YOURLS/YOURLS/discussions).
- [X] I checked [current issues](https://github.com/YOURLS/YOURLS/issues) and this request isn't a duplicate of an existing issue, opened or closed.

### Self troubleshooting

- [X] I've read the [documentation](https://yourls.org/) and made sure to follow it.
- [X] I've read the [troubleshooting first steps](https://github.com/YOURLS/YOURLS/wiki/Troubleshooting-first-steps) guide.

### Version

1.8.2

### Description

Unable to run yourls from a subfolder (/y) keeping the short url like domain.com/XXX

When following wiki instructions, I get ERR_TOO_MANY_REDIRECTS in my main page,

Yourls admin page (domain.com/y/admin) is properly shown, but if I create a redirect (i.e: domain.com/g pointing to google.com), it points to domain.com/y/admin/google.com

### Expectation

My main site should be shown properly and the redirections should work.

### Reproduction steps

1) Installed yourls in root webserver folder.
2) Created the following .htaccess file in the root folder:

```
# BEGIN YOURLS
<IfModule mod_rewrite.c>
RewriteEngine On
RewriteBase /
RewriteCond %{HTTPS} !=on
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^.*$ /yourls-loader.php [L]
</IfModule>
# END YOURLS
```

3) Moved everything to /y
4) Created the following /y/.htaccess file:

```
<IfModule mod_rewrite.c>
RewriteEngine on
RewriteCond %{HTTP_HOST} domain.com$
RewriteCond %{REQUEST_URI} !^/y/
RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d
RewriteRule ^(.*)$ /y/$1
RewriteCond %{HTTP_HOST} domain.com$
RewriteRule ^(/)?$ y/yourls-loader.php [L] 
</IfModule>
```

### Context

Server version: Apache/2.4.41 (Ubuntu)
Server built:   2022-01-05T14:49:56
PHP 7.4.3 (cli) (built: Nov 25 2021 23:16:22) ( NTS )
Server version: 10.3.32-MariaDB-0ubuntu0.20.04.1 Ubuntu 20.04

**Comment by ozh** [comment]
https://github.com/YOURLS/YOURLS/wiki/.htaccess

**Comment by ehguille** [comment]
Hi @ozh,

With all due respecto. I appreciate your work and think yourls used to be a great software. However, don't you think your acting might be consider as disrespectful?

I have already RTFM and followed it step by step. In fact, I recall to have read about some users that have the same issue with 1.8.2. 

Have you even try to reproduce this issue?

Regards,
Guillermo.

**Comment by LeoColomb** [comment]
Thanks for your reply @ehguille.
I understand @ozh's answer can be frustrating, but the "disrespect" has been initiated in your report, where you checked both:

> * I've read the documentation and made sure to follow it.
> * I checked current issues and this request isn't a duplicate of an existing issue, opened or closed.

This topic is widely documented and has been answered quite a lot of time.
Just a quick search for [`subfolder`](https://github.com/YOURLS/YOURLS/issues?q=subfolder) give a good list of answered tickets. The issue on these tickets is always `.htaccess` misconfiguration, thus the link from @ozh answer.

I hope this helps.

**Comment by ehguille** [comment]
Thanks, @LeoColomb. I've opened an issue is because I have already read all documentation I found, even the closed issues, and try several configurations.

Anyway, there's no point on discuss this. I guess is just a matter of empathy.

Regards,
Guillermo.

**Comment by ozh** [comment]
I do lack empathy, I admit ;-)

Your .htaccess as posted in your issue doesn't match the one in the wiki, so ...

**Comment by ehguille** [comment]
Hi again, ozh!

I reinstalled yourls on my server so I could make your swallow your words 😠.

I carefully follow the guide step-by-step, as I did the previous time, and wrote the same .htaccess files (with the only exception that this time i gave the installation folder the proper permission so it create .htaccess by itself). This time, the application did not include the ```RewriteCond %{HTTPS} !=on``` line in the installation folder .htaccess. This is the only difference (in fact, I wrote it because I saw it here https://github.com/YOURLS/YOURLS/wiki/.htaccess).

I also checked that the root folder .htaccess file is exactly equal to the one I posted first.

Given that, and with all the pain in my hearth, I should apologize 🤦🏽‍♂️.

My mistake was that I was accessing to _domain.com/yourlsdirectory/admin_, and generating the short urls from there. If you try so, yo get a "too many redirects" error. When doing so from _domain.com/admin_ the issue dissapears.

To sum up: I still think my .htaccess where ok, but the issue was due to my mistake using the app, not because the .htaccess file (I belive). Anyway, if you could show me where the mistake was, I would really apreciate it.
